tests/slsa_analyzer/checks/test_registry.py

```diff
@@ -8,12 +8,66 @@
 from unittest import TestCase
 from unittest.mock import patch
 
+import pytest
 from hypothesis import given
 from hypothesis.strategies import SearchStrategy, binary, booleans, integers, lists, none, one_of, text, tuples
 
 from macaron.slsa_analyzer.checks.base_check import BaseCheck
 from macaron.slsa_analyzer.checks.check_result import CheckResultType
-from macaron.slsa_analyzer.registry import Registry
+from macaron.slsa_analyzer.registry import CheckCircularDependency, CheckRegistryError, Registry
+
+
+# pylint: disable=protected-access
+@pytest.fixture(name="check_registry")
+def check_registry_fixture() -> Registry:
+    """Return a registry instance with sample checks registered.
+
+    Returns
+    -------
+    Registry
+        The sample registry instance.
+    """
+    # Refresh Registry static variables before each test case
```

Comment:
Should I put a visual representation of the checks as a tree to make it easier for us to understand it?